### ISSUE TITLE
Add support for SAML (Single Sign-On)

### ISSUE THREAD

**Issue by green2k** [metabase/metabase#4308]
It would be great, if the SAML will be supported, as it'll be much more easier to deploy Metabase into a company's infrastructure. 

⬇️  Please click the 👍 reaction instead of leaving a `+1` or 👍  comment

**Comment by kgeis** [comment]
As a university employee, I'd like to add that it's not just companies that use SAML.

**Comment by jpsandiego** [comment]
Looks like they are adding this into the Enterprise edition which is woefully under-documented:

* https://www.metabase.com/docs/v0.30.4/administration-guide/16-authenticating-with-saml.html

> Authenticating with SAML
> This feature is available in the Enterprise edition of Metabase

> Connecting Metabase to your SAML identity provider lets your team access Metabase with ease through SSO.

**Comment by orware** [comment]
Could we have an update on this one?

Is it something that would only be available under an Enterprise Plan or is it something that will be coming to the open source product?

While this might be a good way to differentiate the open source version vs. the enterprise version, it would be great to have it available in the open source product or make it in such a way that it could be dropped in as a plugin of some sort (and sell the plugin to those that may need that specific functionality).

We'll be able to get by ok with the existing LDAP setup that was added, but we've been evolving towards a SSO setup over the past 1.5 years or so and will be lighting up a lot more SAML stuff over the next month so it would be wonderful to be able to include Metabase in the list of software we have integrated with SSO/SAML.
